_drafts/circle-ci-chibios.md

```diff
@@ -0,0 +1,438 @@
+# Extract dependencies
+$ cd ChibiOS/ext
+$ 7za x fatfs-0.13c_patched.7z
+# Build the example
+$ cd ../demos/STM32/RT-STM32F103-STM3210E_EVAL-FATFS-USB
+$ make -j 
+```
+
+Which should give us:
+```shell
+[...]
+Linking build/ch.elf
+Creating build/ch.hex
+Creating build/ch.bin
+Creating build/ch.dmp
+
+   text	   data	    bss	    dec	    hex	filename
+  50450	    228	  98072	 148750	  2450e	build/ch.elf
+Creating build/ch.list
+
+Done
+```
+
+> Note: `7za` is the command line utility for 7zip, which is available for all platforms.  
+
+We need administrative access to the project on Github to setup CI on it, so you’ll want to fork the ChibiOS repository under your profile or organization (you can find our fork at  [https://github.com/memfault/ChibiOS](https://github.com/memfault/ChibiOS) ).
+
+### Creating a CircleCI Project
+
+We’re now ready to set up CircleCI. Head to circleci.com, create an account, select “Add Projects” in the sidebar, and search for your ChibiOS repository.
+
+![](img/circle-ci/search_chibios.png)
+
+Click “Set Up Project” and follow the instructions. ChibiOS is a C project with robust tooling to build it on Linux, so we recommend you select “Linux” as the operating system, and “Other” as the language.
+
+![](img/circle-ci/setup_project.png)
+
+### CircleCI Hello World
+Now that we have a project set up, we need to tell our CI system how to build and test our firmware. In Circle CI this is done by defining a set of **Steps**, **Jobs**, and **Workflows**.
+
+![](img/circle-ci/Diagram-v3--Default.png)
+
+Simply stated, a **Step** is a single command to be run as part of your build and test process. It is the unit of work of our CI system. A **Job** is a collection of steps executed sequentially in a specific environment. A **Workflow** is a collection of jobs executed either sequentially or in parallel based on more complex logic.
+
+You can read more about Steps, Jobs, and Workflows in [CircleCI’s documentation](https://circleci.com/docs/2.0/jobs-steps/#section=getting-started).
+
+For our hello world example, we will have a single job, with the following steps:
+1. Check out the code repository
+2. Print “Hello, world”
+3. Print the current date and time
+
+This is a trivial thing to do with a bash script:
+```bash
+$ git clone git@github.com:memfault/ChibiOS.git
+$ git checkout <commit-to-build>
+$ echo Hello, world
+$ date
+```
+
+Circle CI doesn’t use bash scripts, but instead defines steps, jobs, and workflows via a custom YAML file which should be placed at the root of your repository under `.circleci/config.yml`.
+
+The subset of the syntax we’ll need here is pretty simple:
+```yaml
+# not a valid CircleCI file
+version: 2
+jobs:
+	<job>:
+		<environment>
+		steps:
+			- run:
+				name: <name>
+				command: <shell command>
+			- run:
+				name: <name>
+				command: <shell command>
+	<job>:
+		...
+```
+
+We can fill our steps by simply copying the commands from our bash script
```

Comment:
We can populate our `steps:` by simply copying the shell commands

or something like that. Was hard to follow what you were saying or doing.